web/src/filter.js

```diff
@@ -567,8 +590,9 @@ export class Filter {
     is_common_narrow() {
         // can_mark_messages_read tests the following filters:
         // stream, stream + topic,
-        // is:dm, dm,
-        // is:mentioned, is:resolved
+        // is: dm, dm,
+        // ( -is: dm, stream + -is:dm, topic + -is:dm )
```

Comment:
Thank you for these insights. Yes I would love to hear what the next step should be. 

`initial_filters_for_mark_messages_read() {
        const term_types = this.sorted_term_types();

        if (_.isEqual(term_types, ["stream", "topic"])) {
            return true;
.....


    calc_can_mark_messages_read() {
        // Arguably this should match supports_collapsing_recipients.
        // We may want to standardize on that in the future.  (At
        // present, this function does not allow combining valid filters).

        if (this.initial_filters_for_mark_messages_read()) {
            return true
        }
....
`

Should we add a method like this? Which combines the filters that work with `is_common_narrow()` and we extend it with extra filters in `_can_mark_message_read()`

Or maybe we do it differently, fixing problems with is_common_narrow() and everything else. Let me know.

Reply:
My instinct is that you'll need to do what the TODO comment indicates for `is_common_narrow` and rewrite that so that it does not depend on the logic of `can_mark_messages_read`.

That would be a separate prep commit to the changes you've done here, and it would probably be a good idea to start a conversation about that in the [#frontend](https://chat.zulip.org/#narrow/stream/6-frontend) stream on [chat.zulip.com](https://chat.zulip.org/).